tools/build_api.py

```diff
@@ -233,18 +233,22 @@ def find_valid_toolchain(target, toolchain):
     toolchain_names = get_valid_toolchain_names(target, toolchain)
     last_error = None
     for index, toolchain_name in enumerate(toolchain_names):
-        internal_tc_name = get_toolchain_name(target, toolchain_name)
-        if toolchain == "ARM" and toolchain_name == "ARMC5" and index != 0:
-            end_warnings.append(ARMC5_MIGRATION_WARNING)
+        internal_tc_name = get_toolchain_name(target, toolchain_name)                  
         if not TOOLCHAIN_CLASSES[internal_tc_name].check_executable():
             search_path = TOOLCHAIN_PATHS[internal_tc_name] or "No path set"
             last_error = (
                 "Could not find executable for {}.\n"
                 "Currently set search path: {}"
             ).format(toolchain_name, search_path)
         else:
-            if toolchain_name == "uARM" or target.default_toolchain == "uARM":
-                 end_warnings.append(UARM_TOOLCHAIN_WARNING)
+            if toolchain_name in ["uARM", "ARMC5", "ARMC6"]:
```

Comment:
You can use `elif (condition): pass` instead of `else: if (condition): pass`

Reply:
The reason is, it will fail for GCC_ARM

Reply:
It will not fail. It will work the exact same way. It is weird to have else and and single if.

Reply:
until ```if toolchain_name in ["uARM", "ARMC5", "ARMC6", "GCC_ARM"]:```, it will not work when I combined as if with else.

Reply:
That is not what I originally asked. Read again.

Reply:
I was confused. I know understand that line 244 is only applicable to line 247 (To prevent adding the warning if a target has `target.default_toolchain` set to `uARM` even if say the `GCC_ARM` toolchain is used).
Perhaps line 244 should be moved to above 247 to be clear.